public/New-DbaDbUser.ps1

```diff
@@ -69,165 +74,246 @@ function New-DbaDbUser {
+        ### If this fails, we notify the user and continue with the next instance.
+        ### The next six lines are (nearly) always the same for every command that connects to one or more instances.
         foreach ($instance in $SqlInstance) {
-            #prepare parameter values
-            $connParam.SqlInstance = $instance
-            $getDbParam = $connParam.Clone()
-            $getDbParam.OnlyAccessible = $True
-            if ($Database) {
-                $getDbParam.Database = $Database
-            }
-            if (-not $IncludeSystem) {
-                $getDbParam.ExcludeSystem = $True
-            }
-            if ($ExcludeDatabase) {
-                $getDbParam.ExcludeDatabase = $ExcludeDatabase
+            try {
+                $server = Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
             }
 
-            # Is the login exist?
-            if ($Login -and (-not(Get-DbaLogin @connParam -Login $Login))) {
-                Stop-Function -Message "Invalid Login: [$Login] is not found on [$instance], skipping." -Target $instance -Continue -EnableException $False
+            ### Run checks as early as possible.
+            ### After connecting to the instance, run checks that need a connected instance.
+            ### As the check might be successful on the next instance in the loop, use -Continue.
+            ### Discussion: What is the correct -Target that we want to use. In this case it might also be $server or $Login. Or we just not use it anymore - because: Have we ever used the data?
+            ### Just found a big problem with -Target: If the target is an SMO (example: $db later in the code), the background runspace is constantly querying the instance.
+            ### In the messages, all strings should be surrounded by "[]", but all SMO variables will get "[]" automaticaly by their .ToString() method.
+            if ($Login -and -not $server.Logins[$Login]) {
+                Stop-Function -Message "Login [$Login] not found on $server" -Continue
             }
 
-            $databases = Get-DbaDatabase @getDbParam
-            $getValidSchema = Get-DbaDbSchema -InputObject $databases -Schema $DefaultSchema -IncludeSystemSchemas
-
-            #This block is required so that correct error message can be returned to the user when incorrect database name is given or the database doesn't exists in the server.
-            if (-not $Database) {
-                $Database = $databases.Name
+            ### As we need the database object(s) to be able to add a new users to it, we have to filter the databases of the instance based on the provided parameters.
+            ### We don't use Get-DbaDatabase here, because that command does way more than we need.
```

Comment:
this was an "open point" which I'm not sure I agree upon. What does get-dbadatabase do on top of "the smo thing" that makes it so unuseful/slow ? also .... I recall that on case-sensitive instances SMO had "problems".

Reply:
I just tested this and had a look at the code on Get-DbaDatabase. You are right, there is not much additional work.

But I see one central problem: `Get-DbaDatabase -Database DoesNotExist, master` just returns the master database SMO and does not inform the user that the requested database `DoesNotExist` is not found on the instance. For a command like New-DbaDbUser I would like to fail if the database is not there instead of having to test the output of the command against the database list.

Maybe we should change `Get-DbaDatabase` and then use it in every other command. I'm open to do so.

Reply:
I opened an issue for that: https://github.com/dataplat/dbatools/issues/9387

We can discuss the change there and in case we want a change, we can link the PR to that issue.

Reply:
I changed the code to use Get-DbaDatabase.